autogpt_platform/backend/custom_migrations/001_add_user_trigger.sql

```diff
@@ -0,0 +1,12 @@
+CREATE OR REPLACE FUNCTION add_user_to_platform() RETURNS TRIGGER AS $$
+BEGIN
+    INSERT INTO platform."User" (id, email, "updatedAt")
+    VALUES (NEW.id, NEW.email, now());
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql SECURITY DEFINER;
+
+CREATE TRIGGER user_added_to_platform
+AFTER INSERT ON auth.users
+FOR EACH ROW
+EXECUTE FUNCTION add_user_to_platform();
```

Comment:
Any reason for using a db trigger instead of a python code?

Reply:
what's the benefit of python code here?

Reply:
Yes, the db triggers when Supabase has a user inserted in TS land. Currently this is happing in Python when we attempt to use the user for the first time and it’s causing a race condition of using the user before the Python code is run. Putting a Python call in that function was an option that imo was worse user experience and less clear

Reply:
> race condition
This needs to be within db transaction, in the same manner of transaction.

I meant application code. It doesn't have to be Python, TS works too depending on where the insert is done.

Reply:
Then we lose almost all the benefits of Supabase a user creation. It is also the recommend method https://supabase.com/docs/guides/auth/managing-user-data#accessing-user-data-via-api

Reply:
Wdym by losing all the benefits ? What benefits and why are we losing it my moving the trigger?

I'm just curious why we are addin the migration script and all this change for a stored proceduce. Any clarity on this might be beneficial for others too.

Reply:
Tried moving this to TS land and our schemas aren’t configured correctly to accept this so they aren’t exposed to the TS Supabase api and throw obtuse errors